sdk/identity/identity/src/msal/nodeFlows/msalClientCertificate.ts

```diff
@@ -46,59 +46,70 @@ interface CertificateParts {
 }
 
 /**
- * MSAL client certificate client. Calls to MSAL's confidential application's `acquireTokenByClientCredential` during `doGetToken`.
+ * Tries to load a certificate from the given path.
+ *
+ * @param certificatePath - Path to the certificate.
+ * @param sendCertificateChain - Option to include x5c header for SubjectName and Issuer name authorization.
+ * @returns - The certificate parts, or `undefined` if the certificate could not be loaded.
  * @internal
  */
-export class MsalClientCertificate extends MsalNode {
-  private sendCertificateChain?: boolean;
-
-  constructor(options: MSALClientCertificateOptions) {
-    super(options);
-    this.requiresConfidential = true;
-    this.sendCertificateChain = options.sendCertificateChain;
-
-    const parts = this.parseCertificate(options.certificatePath);
-    this.msalConfig.auth.clientCertificate = {
-      thumbprint: parts.thumbprint,
-      privateKey: parts.certificateContents,
-      x5c: parts.x5c
-    };
+export function parseCertificate(
+  certificatePath: string,
+  sendCertificateChain?: boolean
+): CertificateParts {
+  const certificateParts: Partial<CertificateParts> = {};
+
+  certificateParts.certificateContents = readFileSync(certificatePath, "utf8");
```

Comment:
why are we using `readFileSync`? Why not promisify `readFile` and await? It feels bad to block on this